cmd/rpcdaemon/commands/eth_callMany.go

```diff
@@ -0,0 +1,275 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	timeoutMilliSeconds := int64(5000)
+	if timeoutMilliSecondsPtr != nil {
+		timeoutMilliSeconds = *timeoutMilliSecondsPtr
+	}
+	timeout := time.Millisecond * time.Duration(timeoutMilliSeconds)
+	// Setup context so it may be cancelled the call has completed
+	// or, in case of unmetered gas, setup a context with a timeout.
+	var cancel context.CancelFunc
+	if timeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+	// Make sure the context is cancelled when the call has completed
+	// this makes sure resources are cleaned up.
+	defer cancel()
+
+	// Wait for the context to be done and cancel the evm. Even if the
+	// EVM has finished, cancelling may be done (repeatedly)
+	go func() {
+		<-ctx.Done()
+		evm.Cancel()
+	}()
+
+	// Setup the gas pool (also for unmetered requests)
+	// and apply the message.
+	gp := new(core.GasPool).AddGas(math.MaxUint64)
+	for _, txn := range intraTransactions {
+		msg, err := txn.AsMessage(*signer, nil, rules)
+		if err != nil {
+			return nil, err
+		}
+		// Execute the transaction message
+		_, err = core.ApplyMessage(evm, msg, gp, true /* refunds */, false /* gasBailout */)
+		if err != nil {
+			return nil, err
+		}
+		// If the timer caused an abort, return an appropriate error message
+		if evm.Cancelled() {
+			return nil, fmt.Errorf("execution aborted (timeout = %v)", timeout)
+		}
+	}
+
+	ret := make([][]map[string]interface{}, 0)
+	overrideBlockHash := make(map[uint64]common.Hash)
+
+	for _, bundle := range bundles {
+		// first change blockContext
+		if bundle.BlockOverride.BlockNumber != nil {
+			blockCtx.BlockNumber = uint64(*bundle.BlockOverride.BlockNumber)
+		}
+		if bundle.BlockOverride.BaseFee != nil {
+			blockCtx.BaseFee = bundle.BlockOverride.BaseFee
+		}
+		if bundle.BlockOverride.Coinbase != nil {
+			blockCtx.Coinbase = *bundle.BlockOverride.Coinbase
+		}
+		if bundle.BlockOverride.Difficulty != nil {
+			blockCtx.Difficulty = bundle.BlockOverride.Difficulty
+		}
+		if bundle.BlockOverride.Timestamp != nil {
+			blockCtx.Time = uint64(*bundle.BlockOverride.Timestamp)
+		}
+		if bundle.BlockOverride.GasLimit != nil {
+			blockCtx.GasLimit = uint64(*bundle.BlockOverride.GasLimit)
+		}
+		if bundle.BlockOverride.BlockHash != nil {
+			for blockNum, hash := range *bundle.BlockOverride.BlockHash {
+				if err != nil {
+					return nil, err
+				}
+				overrideBlockHash[blockNum] = hash
+			}
+			blockCtx.GetHash = func(i uint64) common.Hash {
+				if hash, ok := overrideBlockHash[i]; ok {
+					return hash
+				}
+				hash, err := rawdb.ReadCanonicalHash(tx, i)
+				if err != nil {
+					log.Debug("Can't get block hash by number", "number", i, "only-canonical", true)
+				}
+				return hash
+			}
+		}
+		evm = vm.NewEVM(blockCtx, txCtx, evm.IntraBlockState(), chainConfig, vm.Config{Debug: false})
```

Comment:
Could you explain what txCtx is used for? This txCtx is still the firstMsg txCtx. Could that be a problem?

Reply:
txCtx is for tracking the origin of a certain transaction, so we probably want to initialize a new evm for each transaction.

Reply:
I see. Then we probably want to call `transactions.GetEvmContext` for each transaction?